packages/react-scripts/scripts/eject.js

```diff
@@ -36,6 +37,31 @@ prompt(
     process.exit(1);
   }
 
+  // Make sure there are no dirty git status
+  const git = fs.existsSync(path.join(process.cwd(), '.git'));
```

Comment:
@milocosmopolitan someone might have their git directory several levels up the tree, especially if it's a monorepo.

However, you shouldn't be checking for the existence of `.git` at all -- simply execute a git command in the directory and see if it errors out 😄.